anchor-service/src/time_event.rs

```diff
@@ -0,0 +1,162 @@
+use anyhow::{anyhow, Result};
+use cid::Cid;
+
+use ceramic_event::{
+    anchor::{AnchorRequest, TimeEvents},
+    unvalidated::RawTimeEvent,
+};
+
+use crate::DetachedTimeEvent;
+
+pub fn build_time_events(
+    anchor_requests: &[AnchorRequest],
+    detached_time_event: &DetachedTimeEvent,
+    count: u64,
+) -> Result<TimeEvents> {
+    let events = anchor_requests
+        .iter()
+        .enumerate()
+        .map(|(index, anchor_request)| {
+            build_time_event(
+                &anchor_request.id,
+                &anchor_request.prev,
+                &detached_time_event.proof,
+                detached_time_event.path.as_str(),
+                index.try_into()?,
+                count,
+            )
+        })
+        .collect::<Result<Vec<_>>>()?;
+
+    Ok(TimeEvents { events })
+}
+
+pub fn build_time_event(
+    id: &Cid,
+    prev: &Cid,
+    proof_cid: &Cid,
+    remote_path: &str,
+    index: u64,
+    count: u64,
+) -> Result<RawTimeEvent> {
+    let local_path = index_to_path(index, count)?;
+    Ok(RawTimeEvent {
```

Comment:
Why do we not use the builder API here?

Reply:
We have a proof object CID and the builder API takes the fields. We would need to parse the proof block and plum through the fields from the CAS response just to eventually re assemble in the builder api and calculate the same CID. Assuming it round trips witch it should.